src/components/inputs/ModalAddContact.tsx

```diff
@@ -43,8 +56,41 @@ const ModalAddContact = () => {
     }
   }, [email, username]);
 
+  const handlePressSend = async () => {
+    setLoading(true);
+    try {
+      const newContactAdded = await addContact(email);
+      if (newContactAdded) {
+        setContacts(prev => [...prev, newContactAdded]);
+        setLoading(false);
+        setShowModal(false);
+      } else {
+        setShowAlert(true);
+      }
+    } catch (error) {
+      console.log(
+        'Location: components/inputs/ModalAddContacts handlePressSend()',
+        error,
+      );
+    } finally {
+      setEmail('');
+      setUsername('');
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    let time = setTimeout(() => {
```

Comment:
maybe you need to add a if(showAlert) ?
Otherwise is being triggered at the begining too (even if nothing is happening visually)